src/block_cache/block_chunkcache.c

```diff
@@ -28,28 +65,35 @@ static int
 __chunkcache_alloc(WT_SESSION_IMPL *session, WT_CHUNKCACHE_CHUNK *chunk)
 {
     WT_CHUNKCACHE *chunkcache;
-    WT_DECL_RET;
+    size_t bit_index;
+    uint8_t *map_byte;
 
     chunkcache = &S2C(session)->chunkcache;
+    bit_index = 0;
 
     if (chunkcache->type == WT_CHUNKCACHE_IN_VOLATILE_MEMORY)
-        ret = __wt_malloc(session, chunk->chunk_size, &chunk->chunk_memory);
+        WT_RET(__wt_malloc(session, chunk->chunk_size, &chunk->chunk_memory));
     else {
-#ifdef ENABLE_MEMKIND
-        chunk->chunk_memory = memkind_malloc(chunkcache->memkind, chunk->chunk_size);
+retry:
+        /* Use the bitmap to find the free chunks in the cache. */
+        WT_RET(__bitmap_find_free(session, &bit_index));
+
+        /* Mark the free chunk in the bitmap to in use. */
+        map_byte = (uint8_t *)&chunkcache->bitmap[bit_index / 8];
+        if (!__wt_atomic_cas8(map_byte, *map_byte, *map_byte | (uint8_t)(0x01 << (bit_index % 8))))
+            goto retry;
+
+        /* Allocate the free memory in the chunk cache. */
+        chunk->chunk_memory = chunkcache->memory + chunkcache->chunk_size * bit_index;
         if (chunk->chunk_memory == NULL)
```

Comment:
I failed to find a scenario where chunk->chunk_memory would be NULL, hence I was wondering the significance of this if condition.

Reply:
Yup good point. [690e97d](https://github.com/wiredtiger/wiredtiger/pull/9442/commits/690e97db34d0ac5088874a15cf526129f4065290)